include/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+#define _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+
+#include "nabla.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace CentralLimitBoxBlur
+{
+
+typedef uint32_t uint;
+struct alignas(16) uvec4
+{
+    uint x, y, z, w;
+};
+#include "nbl/builtin/glsl/ext/CentralLimitBoxBlur/parameters_struct.glsl"
+
+class CBlurPerformer final : public core::IReferenceCounted
+{
+public:
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t DEFAULT_WORKGROUP_SIZE = 256u;
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t PASSES_PER_AXIS = 3u;
+
+    typedef nbl_glsl_ext_Blur_Parameters_t Parameters_t;
+
+    struct DispatchInfo_t
+    {
+        uint32_t wg_count[3];
+    };
+
+    CBlurPerformer(video::IVideoDriver* driver, uint32_t maxDimensionSize, bool useHalfStorage);
+
+    static core::SRange<const asset::SPushConstantRange> getDefaultPushConstantRanges();
+
+    static inline void defaultBarrier()
+    {
+        video::COpenGLExtensionHandler::extGlMemoryBarrier(GL_SHADER_STORAGE_BARRIER_BIT);
+    }
+
+    static inline size_t getOutputBufferSize(const asset::VkExtent3D& inputDimensions, const uint32_t channelCount)
+    {
+        return inputDimensions.width * inputDimensions.height * inputDimensions.depth * channelCount * sizeof(float);
+    }
+
+    static inline uint32_t buildParameters(uint32_t numChannels, const asset::VkExtent3D& inputDimensions, Parameters_t* outParams, DispatchInfo_t* outInfos,
+        const float radius, const asset::ISampler::E_TEXTURE_CLAMP* wrappingType, const asset::ISampler::E_TEXTURE_BORDER_COLOR* borderColor = nullptr)
```

Comment:
document what unit the radius is measured in

turn this into a `Parameters_t` constructor

Reply:
> document what unit the radius is measured in

Sure, but it is measure in NDC space, isn't it?

> turn this into a Parameters_t constructor

`buildParameters` also builds the `DispatchInfo_t` --add constructor there as well?

Reply:
> Sure, but it is measured in NDC space, isn't it 
I don't remember anymore, please deduce, could be normalized tex coord just as well

> buildParameters also builds the DispatchInfo_t --add constructor there as well?

We can probably get rid of `DispatchInfo_t` cause its pointless to keep it around (the way you work out dispatch sizes is really simple),ca just have a `dispatchHelper` which takes `passAxis`/`passIndex`

P.S. we still need a function to tell us how many passes we need.